### PR TITLE
PSP-3942: Add the container for activities - that will display activi…

### PR DESCRIPTION
**Pull request by Maheshbabu28**
…ty details

### CODE REVIEW

**Comment by devinleighsmith** [comment]
Unfortunately, this does not address the issue presented by the story. This implementation does not address the technical notes provided with the story, for example, it does not handle multiple resolutions in any way.

This story will likely need to be carried forward, as it is not useable in its current state.

image

**Review by devinleighsmith — CHANGES_REQUESTED** [review]
See comment, this story needs to gracefully handle multiple resolutions, as well as the window being resized during use.

File: frontend/src/features/research/activities/ActivityTray/ActivityTray.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          Activity: General
+          <Styled.CloseButton
+            id="close-tray"
+            title="close"
+            onClick={() => {
+              setShow(false);
+              onClose();
+            }}
+          >
+            X
```
Comment (devinleighsmith) [inline frontend/src/features/research/activities/ActivityTray/ActivityTray.tsx:34]: please use an mdclose icon instead.

Comment (devinleighsmith) [reply frontend/src/features/research/activities/ActivityTray/ActivityTray.tsx:34]: @Maheshbabu28

**Comment by devinleighsmith** [comment]
@Maheshbabu28 please demo this to bobbi and get confirmation that what you've implemented here is acceptable.

File: frontend/src/features/research/activities/ActivityTray/styles.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export const TrayHeader = styled.h3`
+  font-size: 2rem;
+  padding: 0.8rem;
+  border-bottom: 0.4rem solid ${props => props.theme.css.primaryColor};
+  border-top-left-radius: 1rem;
+  border-top-right-radius: 1rem;
+  margin-bottom: 3.2rem;
+  background-color: ${props => props.theme.css.primaryColor};
+  color: white;
+  font-family: '' BCSans ',Fallback,sans-serif';
```
Comment (devinleighsmith) [inline frontend/src/features/research/activities/ActivityTray/styles.ts:12]: this css looks invalid to me.

File: frontend/src/features/research/activities/ActivityTray/styles.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+`;
+
+//place high z-index on grandparent so that negative z-index tray still draws over map(but under navbar parent).
+export const ZIndexWrapper = styled.div`
+  z-index: 1000;
```
Comment (devinleighsmith) [inline frontend/src/features/research/activities/ActivityTray/styles.ts:17]: would prefer that this is removed unless absolutely necessary.

File: frontend/src/features/research/activities/ActivityTray/styles.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }
+`;
+
+export const ActivityTray = styled.div`
+  height: 100%;
+  width: 100%
+  overflow-y: hidden;
+  background-color: white;
+  z-index: 10000;
```
Comment (devinleighsmith) [inline frontend/src/features/research/activities/ActivityTray/styles.ts:47]: same here, remove this please.

File: frontend/src/features/research/activities/ActivityTray/styles.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  border-radius: 1rem;
+  text-align: left;
+  transition: transform 0.5s ease-in-out;
+  box-shadow: 0.3rem 0 0.4rem rgba(0, 0, 0, 0.2);
```
Comment (devinleighsmith) [inline frontend/src/features/research/activities/ActivityTray/styles.ts:51]: given your other changes I don't think this is visible anymore.

File: frontend/src/features/properties/map/MapActionWindow.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import styled from 'styled-components';
+
+interface IMapSideBarProps {
+  showWindow: boolean;
+}
+
+const MapActionWindow: React.FunctionComponent<IMapSideBarProps> = ({ showWindow, ...props }) => {
+  return (
+    <StyledMapWindow show={showWindow} data-test="map-window">
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/MapActionWindow.tsx:9]: what is data-test?

File: frontend/src/features/research/activities/ActivityTray/ActivityTray.tsx (hunk 1 above)
Comment (Maheshbabu28) [reply frontend/src/features/research/activities/ActivityTray/ActivityTray.tsx:34]: done

File: frontend/src/features/research/activities/ActivityTray/styles.ts (hunk 5 above)
Comment (Maheshbabu28) [reply frontend/src/features/research/activities/ActivityTray/styles.ts:51]: yes, removed

File: frontend/src/features/research/activities/ActivityTray/styles.ts (hunk 4 above)
Comment (Maheshbabu28) [reply frontend/src/features/research/activities/ActivityTray/styles.ts:47]: done

File: frontend/src/features/research/activities/ActivityTray/styles.ts (hunk 3 above)
Comment (Maheshbabu28) [reply frontend/src/features/research/activities/ActivityTray/styles.ts:17]: removed

File: frontend/src/features/research/activities/ActivityTray/styles.ts (hunk 2 above)
Comment (Maheshbabu28) [reply frontend/src/features/research/activities/ActivityTray/styles.ts:12]: font family is removed not required.

File: frontend/src/features/properties/map/MapActionWindow.tsx (hunk 6 above)
Comment (Maheshbabu28) [reply frontend/src/features/properties/map/MapActionWindow.tsx:9]: it should be data-testid, changed

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**